server/mqtt.go

```diff
@@ -952,7 +952,20 @@ func (c *client) mqttParse(buf []byte) error {
 }
 
 func (c *client) mqttTraceMsg(msg []byte) {
-	maxTrace := c.srv.getOpts().MaxTracedMsgLen
+	opts := c.srv.getOpts()
+	maxTrace := opts.MaxTracedMsgLen
+	headersOnly := opts.TraceHeaders
```

Comment:
I don't think this is relevant for MQTT messages since they don't have headers. This is invoked from the parsing of an MQTT client's pub message.